### PR TITLE
Dead code salvage: circuitboard fuse minigame

### PR DESCRIPTION
**Pull request by MadmanMartian**
Was codediving, found this old commented out functionality of the blank circuitboard, thought it could be fun.

Re-adds the circuit minigame to the blank circuitboard, where you have to find the right configuriation of fuses for it to equate to a board.

Added some QoL, such as in-UI feedback for if you have a working combination. If you do, it shows the name of the resulting board, otherwise it shows a red X.

Moved the blank circuitboard to its own DM file, as I intend to advance on them with an advanced blank circuitboard with more fuses, for more potential circuitboard configurations.

Would like review from the original contributor of this minigame, @Kurfursten 

[content]

### CODE REVIEW

**Comment by ghost** [comment]
Horrible rng grind feature for no reason at all adding nothing but pain and suffering. It's like you want everyone to go to science for everything.

**Comment by iFlashYou** [comment]
make it a admin button

**Comment by Kurfursten** [comment]
I really like the game but I realized that getting circuits basically isn't enough of a reward for people since most stuff like fire alarms you'll make one of in a shift. We may find something good for the minigame some day but this wasn't a good fit.

**Comment by ghost** [comment]
It would be worthwhile if it could create any circuit in existence, as a complementary feature to existing soldering if someone wants to gamble.

**Comment by DrSnips** [comment]
It seems like a neat minigame but I don't know if it should be a thing for getting the already bare bones selection of circuits. It could be a neat gamble thing if it could make any circuits at random. Maybe have it triggered by applying a wire/capacitor to start the minigame?

**Comment by MadmanMartian** [comment]
hmm, could work

 - Blank board can be simply soldered to let you choose the basic boards currently available
 - Apply a capacitor and some wire to upgrade the blank, have to solder it correctly to get an advanced board
 - The more advanced the capacitor, the more potential boards are available

Sound good? Nothing is removed then, only added.

**Comment by DrSnips** [comment]
I like that idea a lot. I would prefer any random board but if it has to be a set list when so be it.
Maybe have the potential boards be based on total research values to save you time generating hard lists? A basic capacitor could get you anything with a cumulative value below 6. >=6, 7-8, 9+ for their respective capacitor tier seems ok for now.

**Comment by ghost** [comment]
Locking it behind higher than basic quality capacitors completely defeats the point, since if you're asking science for capacitors you might as well ask for the board instead of gambling for it.

**Comment by DrSnips** [comment]
That's a fair point.

**Comment by MadmanMartian** [comment]
Hmm.

Considering the basic circuits datum's boards will be redundant with the readdition of the preset board, maybe could have RnD available from tier 1?

>any greyshirt can make their own RnD

Likely a bad idea. Has anyone got any suggestions for tier 1 boards?

**Comment by DrSnips** [comment]
To revise what I said before maybe have better capacitors have a better chance at yielding high research boards. 
It would generate a board, check the research sum and then give you the total which is multiplied by 10 to give us a failure chance as F. The capacitor's tier var is then multiplied by 30 to give us a retention chance as R. Then it subtracts R from F as a probability to keep or reroll the board you get. If the ending failure chance is <0 it rounds to 0 and a similar thing if it's >100 to 100. 0<((Research sum×10)-(capacitor tier×30))<100= probability to keep the board design randomly generated.
TLDR
You can roll any board but better capacitors = better chance for "good" boards.

**Comment by I-VAPE-VOX-CLOACA-EVERY-DAY-OF-MY-LIFE** [comment]
Honestly it might be better to save this minigame for something else, like hacking a certain machine or abandoned crates or something
It doesn't make sense to try to make it compete with the soldering iron that was created specifically to replace it, and we're already at a blank on what circuits to give it

**Comment by ghost** [comment]
Giving assistants something creative to do such as their own RND lab is a good idea, since it will give them more things to do that are not tiding around.

Currently exosuit fabricator and RND console are the only two manufacturing machines that are ID locked without an ability to bypass it. This could also de-snowflake those machines. Having multiple ways to achieve the same things is always a good addition to games.

**Comment by brndd** [comment]
Giving assistants a full blown R&D lab is a terrible idea. That shit gets them bagulooses, guns, and all sorts of other dangerous stuff that leads to even worse greytiding. A gimped ghetto version of the R&D system could work, but the existing R&D system isn't really meaty enough to support that IMO.

**Comment by DrSnips** [comment]
I don't think making it an RnD lab creation tool is a good idea for the above reason but also because it just feels odd how the main thing you would make from this is just a better circuit making tool. My hope would be that it allows you to roll something silly like the xenoarch artifact machines or scp 294.

**Comment by LordMyroc** [comment]
This just seems like a bad idea altogether, and I agree with the above sentiment that this is probably best saved for something else entirely. Anything you can already make with this isn't valuable enough to be locked behind a minigame, and anything that WOULD be valuable enough is generally too powerful to be available to anyone with a solder or capacitor.

**Comment by MadmanMartian** [comment]
Then it is settled. Should this minigame be needed again. The PR is here.

For now, it rests.

**Comment by DrSnips** [comment]
@MadmanMartian I thought of an idea for this. Rigsuit upgrades and nearly everything related to them could be from this as well.
